ign_rviz_plugins/src/rviz/plugins/PoseArrayDisplay.cpp

```diff
@@ -0,0 +1,343 @@
+  headLength(0.07), headRadius(0.03), axisLength(0.3), axisRadius(0.03), axisHeadVisible(false)
+{
+  // Get reference to scene
+  this->engine = ignition::rendering::engine("ogre");
+  this->scene = this->engine->SceneByName("scene");
+
+  this->rootVisual = this->scene->CreateVisual();
+  this->scene->RootVisual()->AddChild(this->rootVisual);
+
+  this->mat = this->scene->CreateMaterial();
+  this->mat->SetAmbient(1.0, 0.098, 0.0);
+  this->mat->SetDiffuse(1.0, 0.098, 0.0);
+  this->mat->SetEmissive(1.0, 0.098, 0.0);
+}
+
+////////////////////////////////////////////////////////////////////////////////
+PoseArrayDisplay::~PoseArrayDisplay()
+{
+  std::lock_guard<std::mutex>(this->lock);
+  // Delete visual
+  ignition::gui::App()->findChild<ignition::gui::MainWindow *>()->removeEventFilter(this);
+  this->scene->DestroyVisual(this->rootVisual, true);
+}
+
+////////////////////////////////////////////////////////////////////////////////
+void PoseArrayDisplay::initialize(rclcpp::Node::SharedPtr _node)
+{
+  std::lock_guard<std::mutex>(this->lock);
+  this->node = std::move(_node);
+}
+
+////////////////////////////////////////////////////////////////////////////////
+void PoseArrayDisplay::subscribe()
+{
+  std::lock_guard<std::mutex>(this->lock);
+
+  this->subscriber = this->node->create_subscription<geometry_msgs::msg::PoseArray>(
+    this->topic_name,
+    this->qos,
+    std::bind(&PoseArrayDisplay::callback, this, std::placeholders::_1));
+}
+
+////////////////////////////////////////////////////////////////////////////////
+void PoseArrayDisplay::setTopic(const std::string & topic_name)
+{
+  std::lock_guard<std::mutex>(this->lock);
+  this->topic_name = topic_name;
+
+  this->subscribe();
+
+  // Refresh combo-box on plugin load
+  this->onRefresh();
+}
+
+////////////////////////////////////////////////////////////////////////////////
+void PoseArrayDisplay::setTopic(const QString & topic_name)
+{
+  std::lock_guard<std::mutex>(this->lock);
+  this->topic_name = topic_name.toStdString();
+
+  // Destroy previous subscription
+  this->unsubscribe();
+  // Reset visualization
+  this->reset();
+  // Create new subscription
+  this->subscribe();
+}
+
+////////////////////////////////////////////////////////////////////////////////
+void PoseArrayDisplay::callback(const geometry_msgs::msg::PoseArray::SharedPtr _msg)
+{
+  std::lock_guard<std::mutex>(this->lock);
+  this->msg = std::move(_msg);
+}
+
+////////////////////////////////////////////////////////////////////////////////
+bool PoseArrayDisplay::eventFilter(QObject * _object, QEvent * _event)
+{
+  if (_event->type() == gui::events::Render::kType) {
+    update();
+  }
+
+  return QObject::eventFilter(_object, _event);
+}
+
+////////////////////////////////////////////////////////////////////////////////
+void PoseArrayDisplay::reset()
+{
+//   this->arrow.visual->SetLocalPose(math::Pose3d::Zero);
```

Comment:
Remove these lines or add a TODO

Reply:
I implemented the reset for visuals in aff5843